### DISCUSSION TITLE
Integration of Victron SmartShunt via VE.Direct

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#474] [Q&A] [answered]
Ich habe gerade gesehen, OpenDTUonBattery kann jetzt mit Victron MPTT und Victron SmartShunt kommunizieren 👍 
Super! 👍  Jetzt ist nicht mehr die Pylontech, die einzige Loesung, die SOC Daten liefern konnte. 

Aber, wie schliesse ich verbinde ich beide (MPTT + SmartShunt) mit der OpenDTU?

**Comment by spcqike** [comment]
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles

Der shunt wird als battery.rx im device Profil angegeben, oder?

**Reply by Manos1966** [reply]
Oh man, @spcqike 
die Frage ist "wie schliesse ich verbinde ich **beide** (MPTT + SmartShunt) " 
Ein Kabel, zwei Kabel, Y-Kabel? Beide haben jeweils einen Ve.Direkt Anschluss...
Es gibt kein einziges Bild (bis jetzt)

**Reply by spcqike** [reply]
Einzeln. Jeweils ein zwei adriges kabel. RX und GND. 

Ve.direct ist eine Punkt-zu-Punkt Verbindung von genau 2 Teilnehmern. 

MPPT und shunt sehen sich nicht. Außer beide haben Bluetooth.

**Reply by Manos1966** [reply]
Das heisst fuer die Doofen (wie ich), ich kann zwei "4Pin MiniMicro JST 2.0 PH" Stecker am VIB des ADUM1201 anschliessen?
![Capture2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/f0af539c-049d-4bdc-9f42-63820813ff5a)

Das ist natürlich Perfekt! 👍 

PS. Ich würde MPPT und SmartShunt Versionen mit Bluetooth kaufen und die zwei miteinander kommunizieren lassen. So würde der MPTT die Batterie optimal laden (gehe ich davon aus).

**Reply by schlimmchen** [reply, marked as answer]
> Oh man, @spcqike

Was für einen Grund hast du, dich über seine korrekte Antwort zu echauffieren?

> Das heisst fuer die Doofen (wie ich), ich kann zwei "4Pin MiniMicro JST 2.0 PH" Stecker am VIB des ADUM1201 anschliessen?

Nein, das ist falsch. VE.Direct ist kein Bus, sondern wie schon von @spcqike erwähnt eine Punkt-zu-Punkt-Verbindung.

Wenn du einen Victron MPPT Solarladeregler und einen Victron SmartShunt anschließen willst, dann über zwei getrennte Kanäle. Die Kommunikation mit dem einen Gerät hat nichts mit der mit dem anderen zu tun. Der TX Pin des Ladereglers kommt an einen Pin des ESP32, den du als victron.rx im device profile definierst, und der TX Pin des SmartShunbts kommt an einen Pin des ESP32, den du als battery.rx definierst. Und wenn du einen ADUM1201 verwenden willst, dann brauchst du zwei davon.

Was muss im Wiki stehen, damit das klar wird?

**Reply by schlimmchen** [reply]
@Manos1966 Siehe [Wiki](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface).

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3578416/93f0f9c3-88ff-46a9-900b-88d1645eabf7)

Jetzt klar?

**Reply by Manos1966** [reply]
Oh, danke fuer die Erklaerungen @schlimmchen

Ich war verwirrt, weil ich die Victron TX/Victron RX Erwähnungen der Wiki als victron.tx/victron.rx Pinout interpretiert hatte. Und battery.tx/battery.rx das Pinout des Pylontech CANbus bis jetzt war.

> Connect **Victron TX** (an output) to the input of the "left" voltage domain of the chip, i.e., VIB. Optional: Connect **Victron RX** (an input) to the output of the "left"

Jetzt ist es im Wiki klar, und ich habe auch die Beschreibung im Device Profiles gesehen 👍 
![Pinout Definitions](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/2f2e8f24-cf2c-4690-9e39-2f94488bb0fc)

**Reply by schlimmchen** [reply]
Verstanden. Dann schrieb ich da besser "VE.Direct TX" statt "Victron TX". Das macht es ein wenig besser.

Noch besser wäre es, wenn die Pins im device profile umbenannt würden von "victron" in "mppt" oder sowas, aber das ist nicht möglich ohne breaking changes und manuellem Eingriff eines Benutzers vor oder nach dem Update. Daher lass uns damit leben, dass mit "victron" im device profile ein VE.Direct charge controller gemeint ist. Vielleicht können wir das aufräumen, wenn wir die Unterstützung für mehrere MPPTs dazugewinnen.

> und ich habe auch die Beschreibung im Device Profiles gesehen 👍

Auch das hatte @spcqike schon verlinkt und hätte dich auf die richtige Fährt bringen sollen.

Schön, dass du es jetzt nachvollziehen konntest. Dann schließ doch diese Diskussion bitte als gelöst.

Ich passe das Wiki (Device Profile und VE.Direct) nochmal an, um unser Learning festzuhalten.

**Reply by Manos1966** [reply]
Sorry @spcqike dass ich grantig war... mit battery.rx hatte ich CANbus im Kopf

**Reply by Manos1966** [reply]
> Verstanden. Dann schrieb ich da besser "VE.Direct TX" statt "Victron TX". Das macht es ein wenig besser.

Ich glaube es war eher mein "eingeschränkter Horizon" nach zwei Monate Pylontech Experimente, als deine Beschreibung... ;-)
